src/support/scratch.c

```diff
@@ -274,6 +274,13 @@ __wt_scr_alloc_func(WT_SESSION_IMPL *session, size_t size, WT_ITEM **scratchp
     size_t allocated;
     u_int i;
 
+    /*
+     * To prevent concurrent access to the scratch buffer when multiple connection APIs use the same
```

Comment:
How about adding some more details like
`To prevent concurrent access to the scratch buffer when multiple threads use the same session. For example, when multiple ...`

Reply:
Can you also add a comment on why we are doing it only for the internal sessions? Internals sessions are the ones that can be shared across the threads.

Reply:
Updated the comments in daa65b9d0f6391362fc42b85bbec88c756099436